diff --git a/node_modules/@creamy-dev/passy/index.js b/node_modules/@creamy-dev/passy/index.js
index 5a98afc..2315ebe 100644
--- a/node_modules/@creamy-dev/passy/index.js
+++ b/node_modules/@creamy-dev/passy/index.js
@@ -1,6 +1,7 @@
 const PassProxy = require("./libs/passedWS/index.js");
 const { WebSocketServer } = require("ws"),
-  net = require("net");
+  net = require("net"),
+  dgram = require("dgram");
 
 class PassyServer {
   constructor(options) {
@@ -9,12 +10,12 @@ class PassyServer {
 
   async main() {
     const options = this.options;
-    const wss = new WebSocketServer({ port: options.originalPort });
+    this.wss = new WebSocketServer({ port: options.originalPort });
 
-    wss.on("connection", async function (ws) {
+    this.wss.on("connection", async function (ws) {
       ws.isReady = false;
 
-      const server = new net.Socket();
+      const server = options.server.protocol == "UDP" ? new dgram.Socket("udp4") : new net.Socket();
 
       server.connect(options.server.port, options.server.ip, function () {
         ws.isReady = true;
@@ -38,9 +39,9 @@ class PassyServer {
     delete opts.originalPort;
     delete opts.passedPort;
 
-    const proxy = new PassProxy(opts);
+    this.proxy = new PassProxy(opts);
 
-    await proxy.init();
+    await this.proxy.init();
   }
 }
 
